### PR TITLE
A backend property request

### PR DESCRIPTION
**Pull request by shssf**

### CODE REVIEW

File: src/ngraph/runtime/backend.hpp [hunk 1]
```diff
@@ -124,6 +124,16 @@ class ngraph::runtime::Backend
     /// \returns true if the op is supported, false otherwise.
     virtual bool is_supported(const Node& node) const;
 
+    /// \brief A set of properties supported by a backend
+    enum class Property
+    {
+        memory_attach /// A tansor can attach the user memory
```
Comment (rkimballn1) [inline src/ngraph/runtime/backend.hpp]: You misspelled `tensor`

Comment (shssf) [reply src/ngraph/runtime/backend.hpp]: fixed

**Comment by shssf** [comment]
@rkimballn1 Could you please look into new changes in CPU backend?
If it is ok, this PR ready to merge.

File: src/ngraph/runtime/backend.hpp [hunk 2]
```diff
@@ -124,6 +124,16 @@ class ngraph::runtime::Backend
     /// \returns true if the op is supported, false otherwise.
     virtual bool is_supported(const Node& node) const;
 
+    /// \brief A set of properties supported by a backend
+    enum class Property
+    {
+        memory_attach /// A tensor can attach the user memory
```
Comment (jbobba) [inline src/ngraph/runtime/backend.hpp]: its not clear what this means. Could you perhaps explain in more detail?

File: src/ngraph/runtime/cpu/cpu_backend.cpp [hunk 3]
```diff
@@ -145,3 +145,13 @@ vector<runtime::PerformanceCounter>
     }
     return rc;
 }
+
+bool runtime::cpu::CPU_Backend::is_supported_property(const Property prop) const
+{
+    if (prop == Property::memory_attach)
+    {
+        return true;
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/cpu_backend.cpp]: why is this true? what about the CPU backend makes this property supported. Is there a way to test that a backend does indeed support the property.

File: src/ngraph/runtime/backend.hpp (hunk 2 above)
Comment (shssf) [reply src/ngraph/runtime/backend.hpp]: @jbobba I explained these things in PPT I sent you some time ago. We discussed that.
or maybe you mean a comment in the code is not clear?

File: src/ngraph/runtime/cpu/cpu_backend.cpp (hunk 3 above)
Comment (shssf) [reply src/ngraph/runtime/cpu/cpu_backend.cpp]: @jbobba As far as I see the CPU has memory attach ability as already used in nGraph-TF https://github.com/NervanaSystems/ngraph-tf/blob/master/src/ngraph_encapsulate_op.cc#L316

File: src/ngraph/runtime/backend.hpp (hunk 2 above)
Comment (jbobba) [reply src/ngraph/runtime/backend.hpp]: i dont recall the PPT but would be nice to make the comment here more self-explanatory.

**Pull request merged**